packages/frontend/src/components/common/ReferenceLine.tsx

```diff
@@ -0,0 +1,19 @@
+import { MarkLineData } from '@lightdash/common';
+
+export type ReferenceLineField = {
+    fieldId?: string;
+    data: MarkLineData;
+};
+
+export const getMarkLineAxis = (
+    xField: string | undefined,
+    flipAxes: boolean,
+    fieldId: string,
+): string => {
+    const reverseFlippedAxis = (defaultAxis: string) => {
+        if (flipAxes) return defaultAxis === 'xAxis' ? 'yAxis' : 'xAxis';
+        return defaultAxis;
+    };
+
+    return reverseFlippedAxis(xField === fieldId ? 'xAxis' : 'yAxis');
```

Comment:
Found this logic hard to follow with the internal function and the logic around the string and not the boolean.
Let me know what you think about this:
```
const isDefaultXAxis = xField === fieldId;
if(flipAxes) {
   return isDefaultXAxis ? 'yAxis' : 'xAxis'
}
return isDefaultXAxis ? 'xAxis' : 'yAxis'
```